From 1746a06949656126037cd3ceb6678358f75cec07 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <jmc@joyent.com>
Date: Tue, 22 Nov 2016 17:17:45 -0800
Subject: [PATCH] joyent/manta-thoth#143 handle unexpected errors when fetching
 configuration Reviewed by: Bryan Cantrill <bryan@joyent.com> Approved by:
 Bryan Cantrill <bryan@joyent.com>

---
 bin/thoth    | 11 +++++++++--
 package.json |  2 +-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/bin/thoth b/bin/thoth
index e63e21a..e9bd6c4 100755
--- a/bin/thoth
+++ b/bin/thoth
@@ -3499,8 +3499,15 @@ var configure = function (cb)
 	client = connect();
 
 	client.get(object, function (err, stream, res) {
-		if (err && err.code == 'ResourceNotFound')
-			return (cb(client, 'from source'));
+		if (err) {
+			if (err.code === 'ResourceNotFound') {
+				cb(client, 'from source');
+				return;
+			}
+
+			fatal('could not load thoth configuration (' +
+			    object + ') from Manta: ' + err);
+		}
 
 		var output = '';
 
diff --git a/package.json b/package.json
index 6c0e252..5abd9a2 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "manta-thoth",
 	"description": "Manta-based system for core and crash dump analysis",
-	"version": "0.2.0",
+	"version": "0.2.1",
 	"author": "Joyent (joyent.com)",
 	"dependencies": {
 		"manta": "1.6.0",
-- 
2.21.0

